pkg/nats/stream/producer_client.go

```diff
@@ -132,7 +134,8 @@ func (pc *ProducerClient) heartBeat(ctx context.Context) {
 
 				msg, err := pc.Conn.Request(v, data, pc.config.HeartBeatRequestTimeout)
 				if err != nil {
-					log.Ctx(ctx).Err(err).Msg("error while sending heart beat request from NATS streaming producer client")
+					log.Ctx(ctx).Err(err).Msg("heartbeat request to consumer client timed out")
+					nonActiveStreamIds[c] = append(nonActiveStreamIds[c], activeStreamIds...)
```

Comment:
does this mean we close all streams the first time we fail to heartbeat? There can be an area of improvement here we should only close streams that fail N consecutive heartbeats, where N can equal to 3 by default

Reply:
Makes sense. I already have an issue opened for optimising this ! Will solve it as part of that !  #4026